src/ui/components/auth/SignInContent.tsx

```diff
@@ -0,0 +1,193 @@
+/* Copyright (c) 2022, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useEffect, useState } from "react";
+import { HTTPStatusCodes, StorageKeys } from "../../../constants";
+import { localStorageHandler } from "../../../services/storage";
+import { getApiUrl, getImageUrl, useFetchData } from "../../../utils";
+import { validateEmail } from "../../../utils/form";
+import InputField from "../inputField/InputField";
+
+interface SignInContentProps {
+	onSuccess: () => void;
+	onCreateNewUserClick: () => void;
+	onForgotPasswordBtnClick: () => void;
+}
+
+interface IErrorObject {
+	email: string;
+	password: string;
+}
+
+const SignInContent: React.FC<SignInContentProps> = ({
+	onSuccess,
+	onCreateNewUserClick,
+	onForgotPasswordBtnClick,
+}): JSX.Element => {
+	const [isLoading, setIsLoading] = useState<boolean>(false);
+	const [userTriedToSubmit, setUserTriedToSubmit] = useState(false);
+	const fetchData = useFetchData({ skipErrorBoundary: true });
+
+	const [email, setEmail] = useState("");
+	const [password, setPassword] = useState("");
+
+	const [errors, setErrors] = useState<IErrorObject>({
+		email: "",
+		password: "",
+	});
+	const [serverValidationError, setServerValidationError] = useState("");
+
+	const validateCredentials = async () => {
+		const response = await fetchData({
+			url: getApiUrl("/api/signin"),
+			method: "POST",
+			config: {
+				body: JSON.stringify({
+					email,
+					password,
+				}),
+			},
+		});
+		const body = await response.json();
+		if (response.status === HTTPStatusCodes.OK && body.status === "OK") {
+			localStorageHandler.setItem(StorageKeys.AUTH_KEY, body.sessionId);
+			onSuccess();
+		}
+		if (response.status === HTTPStatusCodes.UNAUTHORIZED) {
```

Comment:
The sign in API can return USER_SUSPENDED_ERROR as well or USER_LIMIT_REACHED_ERROR. Both of these need to be handled

Reply:
Both errors will have a specific error message